guides/assets/javascripts/guides.js

```diff
@@ -102,5 +103,91 @@
       }, 3000);
       e.clearSelection();
     });
+
+    var mainColElems = Array.from(document.getElementById("mainCol").children);
+    var navLinks = document.querySelectorAll("#subCol a");
+    var DESKTOP_THRESHOLD = 1024;
+
+    var matchingNavLink = function (elem) {
+      if(!elem) return;
+      var index = mainColElems.indexOf(elem);
+
+      var match;
+      while (index >= 0 && !match) {
+        var link = mainColElems[index].querySelector("a");
+        if (link) {
+          match = document.querySelector('#subCol [href="' + link.getAttribute("href") + '"]');
+        }
+        index--;
+      }
+      return match;
+    }
+
+    var removeHighlight = function () {
+      for (var i = 0, n = navLinks.length; i < n; i++) {
+        navLinks[i].classList.remove("active");
+      }
+    }
+
+    var updateHighlight = function (elem) {
```

Comment:
I've noticed that the update highlight logic triggers often as you scroll, even within the same section (which makes sense)... But that got me thinking: should we only trigger all of the highlight logic if the actual element has changed? In other words, if the current element is the one "active", maybe there's no need to remove & re-add the highlight, assume it's correct and skip all that work?